move cent conversions upstream to source package (#101)
fivetran/dbt_stripe: CHANGELOG.md

# dbt_stripe v0.16.0

## Breaking Changes
- This package assumes that amount-based fields, which as raw values are represented in the smallest denomination in Stripe, are cent-based. This PR shifts the existing conversion from cents to dollars to further upstream. Previously, currency-related fields were converted in downstream models, but now have been converted directly in staging models. Since currency-related fields now have different values, this is a breaking change. 

> To disable this default conversion, refer to the [README]((https://github.com/fivetran/dbt_stripe/blob/main/README.md#disabling-cent-to-dollar-conversion)) on disabling the `stripe__amount_divide` variable.

Reviewer: See notes from the source PR to make similar updates here.

Reviewer: Additionally, we should highlight each field in the end models which are seeing this update. This will drastically change the outputs if customers were not previously aware. We should make this obvious to customers.

Author: Would this change the outputs if the values of the fields remain the same? For example, `total_sales` in customer_overview would still retain the same value, just the division occurs farther upstream. I can make a callout for the  fields where the code has changed though. Which, by the way, you make a good point in that the division didn't occur across all end models originally. Will circle back on that in the other comment.

> # dbt_stripe v0.16.0

Reviewer: We'll need to update this based on the new variable being set to false now. We can likely remove the breaking change impacts but mention that the value of this variable will change in the future and they should be made aware, but if they want to set the variable to true then they should know the impact.

Author: downgraded the version to 0.15.2 and made changes to the wording

Author: returning to v0.16.0 and updated the changelog notes

# dbt_stripe v0.15.2

## Feature Update: Optionally convert Amounts to Major Units

> - Stripe passes amount-based fields, such as `amount`, `net`, and `fee`, in the smallest denomination as raw form. This means, if your currency has minor and major units such as USD, 100 represents 100 cents, the minor unit, or 1 USD, the major unit. Alternatively, if your currency doesn't use minor units such as JPY, 100 represents 100 JPY. 

Author: ```suggestion
## Breaking Change
- The aggregated amount-based fields in `stripe__daily_overview` and `stripe__customer_overview` now reflect the raw smallest units (e.g., cents), following Stripe's raw data, instead of converted dollar amounts. This change standardizes values across all models.  

  **Impact**: Customers using these models should note that the values will appear inflated compared to the previous dollar-based representation.  

- Important to note:  
  - The cent-to-dollar conversion has been moved upstream and is now controlled by a new variable: `stripe__convert_values` which is disabled by default.  
  - To enable the conversion and divide all amount-based fields by 100, set `stripe__convert_values` to `True` in your `project.yml`.  
  - See below for more context behind these changes and for detailed setup instructions, see the [README](https://github.com/fivetran/dbt_stripe?tab=readme-ov-file#enabling-cent-to-dollar-conversion).  

## Feature Update: Optional Conversion to Major Units
- **Background**:  Stripe provides amount-based fields (e.g., `amount`, `net`, `fee`) in the smallest denomination. For currencies with minor and major units (e.g., USD), 100 represents 100 cents (1 USD). For currencies without minor units (e.g., JPY), 100 represents 100 JPY.  
```

- This PR introduces a variable `stripe__convert_values` (disabled by default) upstream in the staging models which allows users the option to divide all amount-based fields by 100.

> - If your data is using a currency with major and minor units, it may be useful to enable this variable if you wish to divide amounts by 100 and therefore convert values into major units. For information on how to do so, refer to the [README]((https://github.com/fivetran/dbt_stripe?tab=readme-ov-file#enabling-cent-to-dollar-conversion)) on enabling the `stripe__convert_values` variable.

Author: ```suggestion
- If your data is using a currency with major and minor units, it may be useful to enable this variable if you wish to divide amounts by 100 and therefore convert values into major units. For information on how to do so, refer to the [README]((https://github.com/fivetran/dbt_stripe?tab=readme-ov-file#enabling-cent-to-dollar-conversion)) on enabling the `stripe__convert_values` variable.
  - Examples of currencies using minor units (in which enabling `stripe__convert_values` may be useful) include USD (100 = 1 USD), EUR (100 = 1 Euro), and CAD (100 = 1 CAD).  
```

> - If your Stripe data is *not* using a currency involving minor units, it may make more sense to retain the amount-based fields as raw form and you may run the package without additional configuration.

Author: ```suggestion
- If your Stripe data is *not* using a currency involving minor units, it may make more sense to retain the amount-based fields as raw smallest units by leaving `stripe__convert_values` disabled.
  - Examples of currencies NOT using minor units include JPY (100 = 100 JPY), IDR (100 = 100 IDR), KRW (100 = 100 KRW).
```

- Examples of currencies using minor units (in which enabling `stripe__convert_values` is relevant) include USD, Euro, and CAD.
- Examples of currencies NOT using minor units (in which it makes more sense to keep the amount-based fields in raw form) include Japanese Yen (JPY), Indonesian Rupiah (IDR), and Korean Won (KRW).

## Under the Hood
- Added logic to maintain backwards compatibility in `int_stripe__account_daily` and `stripe__customer_overview` by accounting for the previously existing division by 100. 

> - Updated the `run_models.sh` script to test for when `stripe__convert_values` is set to True.

Author: ```suggestion

```

## Notes
- Currently this package does not support multiple currencies, but we have created a [feature request to support multiple currencies](https://github.com/fivetran/dbt_stripe/issues/102) where you are welcome to provide feedback or contribute to the discussion.

## Documentation
- Updated the descriptions for all amount-based fields to specify the grain of the values and add information about the `stripe__convert_values` variable.
- Added Quickstart model counts to README. ([#103](https://github.com/fivetran/dbt_stripe/pull/103))
- Corrected references to connectors and connections in the README. ([#103](https://github.com/fivetran/dbt_stripe/pull/103))

> 

Author: ```suggestion

## Under the Hood
- Updated the `run_models.sh` script to test for when `stripe__convert_values` is set to True.
- Added a consistency test for `stripe__customer_overview`.

```

> # dbt_stripe v0.15.2

Author: updated

  - Examples of currencies NOT using minor units (in which it makes more sense to keep the amount-based fields in raw form) include Japanese Yen (JPY), Indonesian Rupiah (IDR), and Korean Won (KRW).

## Notes
- This update is not providing multiple currency support; however, we're interested in exploring this functionality if there's a need. As such, we have created a [feature request to support multiple currencies](https://github.com/fivetran/dbt_stripe/issues/102) where you are welcome to provide feedback or contribute to the discussion.

## Under the Hood
- Updated the `run_models.sh` script to test for when `stripe__convert_values` is set to True.
- Added a consistency test for `stripe__customer_overview`.

## Documentation
- Updated the descriptions for all amount-based fields to specify the grain of the values and add information about the `stripe__convert_values` variable.
- Added Quickstart model counts to README. ([#103](https://github.com/fivetran/dbt_stripe/pull/103))
- Corrected references to connectors and connections in the README. ([#103](https://github.com/fivetran/dbt_stripe/pull/103))

> ## Under the Hood
> - Updated the `run_models.sh` script to test for when `stripe__convert_values` is set to True.
> - Added a consistency test for `stripe__customer_overview`.
> 

Reviewer: Looks like a duplicate section that can be removed
```suggestion
```

# dbt_stripe v0.16.0

> ## Breaking Change
> - The aggregated amount-based fields in `stripe__daily_overview` and `stripe__customer_overview` now reflect the raw smallest units (e.g., cents), following Stripe's raw data, instead of converted dollar amounts. This change standardizes values across all models.  
> 
>   **Impact**: Customers using these models should note that the values will appear inflated compared to the previous dollar-based representation.  
> 
> - Important to note:  
>   - The cent-to-dollar conversion has been moved upstream and is now controlled by a new variable: `stripe__convert_values` which is disabled by default.  
>   - To enable the conversion and divide all amount-based fields by 100, set `stripe__convert_values` to `True` in your `project.yml`.  
>   - See below for more context behind these changes and for detailed setup instructions, see the [README](https://github.com/fivetran/dbt_stripe?tab=readme-ov-file#enabling-cent-to-dollar-conversion).  
> 
> ## Feature Update: Optional Conversion to Major Units
> - **Background**:  Stripe provides amount-based fields (e.g., `amount`, `net`, `fee`) in the smallest denomination. For currencies with minor and major units (e.g., USD), 100 represents 100 cents (1 USD). For currencies without minor units (e.g., JPY), 100 represents 100 JPY.  
> 
> - This PR introduces a variable `stripe__convert_values` (disabled by default) upstream in the staging models which allows users the option to divide all amount-based fields by 100.
> 
> - If your data is using a currency with major and minor units, it may be useful to enable this variable if you wish to divide amounts by 100 and therefore convert values into major units. For information on how to do so, refer to the [README]((https://github.com/fivetran/dbt_stripe?tab=readme-ov-file#enabling-cent-to-dollar-conversion)) on enabling the `stripe__convert_values` variable.
>   - Examples of currencies using minor units (in which enabling `stripe__convert_values` may be useful) include USD (100 = 1 USD), EUR (100 = 1 Euro), and CAD (100 = 1 CAD).  
> 
> - If your Stripe data is *not* using a currency involving minor units, it may make more sense to retain the amount-based fields as raw smallest units by leaving `stripe__convert_values` disabled.
>   - Examples of currencies NOT using minor units include JPY (100 = 100 JPY), IDR (100 = 100 IDR), KRW (100 = 100 KRW).
> 
> ## Notes
> - Currently this package does not support multiple currencies, but we have created a [feature request to support multiple currencies](https://github.com/fivetran/dbt_stripe/issues/102) where you are welcome to provide feedback or contribute to the discussion.

Reviewer: Suggestions to consolidate the CHANGELOG since it was becoming a bit repetitive and lengthy while also making sure it's messaging was consistent with what is listed in the source PR.
```suggestion
## Breaking Change
- The aggregated net, gross, sale, charge, refund, and fee amount-based fields in `stripe__daily_overview` and `stripe__customer_overview` now reflect the raw smallest units (e.g., cents), following Stripe's raw data, instead of converted dollar amounts. This change standardizes values across all models.  
  - Customers using these models should note that the values will appear inflated compared to the previous dollar-based representation.  
  - The cent-to-dollar conversion has been moved upstream and is now controlled by a new variable: `stripe__convert_values` which is disabled by default.  
  - See the below Feature Update notes for more context behind these changes and for detailed setup instructions, see the [README](https://github.com/fivetran/dbt_stripe?tab=readme-ov-file#enabling-cent-to-dollar-conversion).  

## Feature Update: Optional Conversion to Major Units
Stripe passes amount-based fields, such as `amount`, `net`, and `fee`, in the smallest denomination as raw form. This means, if your currency has minor and major units such as USD, 100 represents 100 cents, the minor unit, or 1 USD, the major unit. Alternatively, if your currency doesn't use minor units such as JPY, 100 represents 100 JPY. 

- This PR introduces a variable `stripe__convert_values` (disabled by default) upstream in the staging models which allows users the option to divide all amount-based fields by 100.
  - For information on how to enable the division, refer to the [README](https://github.com/fivetran/dbt_stripe?tab=readme-ov-file#enabling-cent-to-dollar-conversion) on configuring the `stripe__convert_values` variable.
  - Otherwise, amount-based fields will be brought through in their raw form.
  - Examples of currencies using minor units (in which enabling `stripe__convert_values` is relevant) include United States Dollar (USD), Euro (EUR), and the Canadian Dollar (CAD).
  - Examples of currencies NOT using minor units (in which it makes more sense to keep the amount-based fields in raw form) include Japanese Yen (JPY), Indonesian Rupiah (IDR), and Korean Won (KRW).

## Notes
- This update is not providing multiple currency support; however, we're interested in exploring this functionality if there's a need. As such, we have created a [feature request to support multiple currencies](https://github.com/fivetran/dbt_stripe/issues/102) where you are welcome to provide feedback or contribute to the discussion.
```
